_posts/2018/2018-03-18-check-your-database-indexes-for-activerecord-scopes.md

```diff
@@ -0,0 +1,59 @@
+---
+title: 'Check Your Database Indexes For ActiveRecord Scopes'
+description: 'Database indexes are always more important than you think'
+layout: article
+category: ruby
+image:
+  base: '2018/check-your-database-indexes-for-activerecord-scopes'
+  alt: 'Construction Site'
+  source: 'https://unsplash.com/photos/8Gg2Ne_uTcM'
+  credit: '贝莉儿 NG'
+---
+
+It is alleged that Ruby is slow. However, whether you're using Ruby, Elixir or Go, the 'benchmarked' speed of a language is irrelevant if your web application eventually hits an unoptimised database.
+
+
+## Instead of…
+
+...having your database do more work than it needs to.
+
+
+## Use…
+
+...database indexes when you query on foreign keys and for queries over multiple fields.
+
+```ruby
+class AddRecommendedIndexes < ActiveRecord::Migration[5.1]
+  def change
+    add_index :model_with_foreign_key, :user_id
+    add_index :polymorphic_model, [:polymorphic_id, :polymorphic_type]
+    add_index :model_found_by_other_field, :slug
+    add_index :model_with_complex_queries, [:slug, :title, :score, :updated_at]
+    # if you're using multiple fields to find & order
```

Comment:
Is this referring to the whole `change` method or just the last index addition?

If the former, move this comment to above line 27, or add text pre the Ruby bit.